### ISSUE TITLE
onGameResize called every time an overlay is added and when game starts

### ISSUE THREAD

**Issue by dbilgin** [flame-engine/flame#1792]
# Current bug behaviour

Whenever a game starts for the first time or whenever `gameRef.overlays.add` is called after for example a level is done, `onGameResize` is called.

# Expected behaviour

For the game start, I think this shouldn't be called as the game is not "resized" but just created for the first time and adding an overlay should simply not trigger a game resize.

# Steps to reproduce

- Create a game
- Add an overlay after the game is launched

# Environment information
* Flame version: ^1.2.0
* Platform affected: android, ios, web
* Platform version affected: seems to be all
* Flutter (Channel stable, 3.0.4, on macOS 12.2.1 21D62 darwin-arm, locale en-DE)

**Comment by spydon** [comment]
Can you verify whether this is fixed now @dbilgin?
`onGameResize` is intentionally called on start, that is part of the lifecycle, but it shouldn't be called when an overlay is added.

**Comment by dbilgin** [comment]
hi @spydon it has been a while I worked on that project, but I updated my flame dependency and tried to run it, but `game.overlays.add` is now giving me an error.

First off, here's my code for the overlays:

```dart
class GameScreen extends StatelessWidget {
  const GameScreen({Key? key, required this.level}) : super(key: key);
  final int level;

  @override
  Widget build(BuildContext context) {
    final game = MahGame(level: level);
    game.overlays.add(Overlays.game.name);

    return GameWidget(
      game: game,
      overlayBuilderMap: {
        Overlays.game.name: (context, MahGame game) =>
            const GameOverlay(),
        Overlays.levelEnd.name: (context, MahGame game) =>
            LevelEndOverlay(level: level),
      },
    );
  }
}
```

And here's the error I am getting:
![image](https://user-images.githubusercontent.com/15243788/214286310-aea327bf-deab-451b-b08f-3c96f47ccd08.png)

And finally, here's the overlay code:

```dart
class GameOverlay extends StatelessWidget {
  const GameOverlay({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Material(
      child: GestureDetector(
        onTapUp: (TapUpDetails details) async {},
        child: Container(
          color: Colors.red,
          child: const Text('Pause'),
        ),
      ),
    );
  }
}
```

So unfortunately, I am not able to test the bug I reported right now, has something changed with the API maybe that I am adding the overlay wrong now?

**Comment by dbilgin** [comment]
Hi, sorry it took a long time, just managed to find time to check this out.
I seem to still be experiencing this. Here are my flame dependencies:
```

  flame: ^1.6.0
  flame_bloc: ^1.6.0
  flame_forge2d: ^0.12.5
```

The game class:

```
class TowerGame extends Forge2DGame with PanDetector, HasTappables {
  TowerDownGame({required this.level});
  final int level;

  @override
  void onGameResize(Vector2 canvasSize) {
    super.onGameResize(canvasSize);
    print('Test');
  }
...
```

And here's how I add the overlay:

```
  Future<void> proceedLevel({required double starSlice}) async {
    final isLevelFinished = await levelStars.setNextStar(starSlice: starSlice);

    if (isLevelFinished) {
      if (levelStars.totalStars == 3.0) {
        await unlockAchievement(GameAchievement.bullseye);
      }

      await HiveDB().addLevelStars(
        level,
        levelStars.totalStars,
        points.currentPoints,
      );
      gameRef.overlays.add(Overlays.levelEnd.name);
    }
  }
```

And finally my overlay:

```
class LevelEndOverlay extends StatefulWidget {
  const LevelEndOverlay({Key? key, required this.level}) : super(key: key);
  final int level;

  @override
  State<LevelEndOverlay> createState() => _LevelEndOverlayState();
}

class _LevelEndOverlayState extends State<LevelEndOverlay> {
  @override
  void initState() {
    submitLeaderboardScore();
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.black.withAlpha(175),
      body: Center(
        child: Column(
          children: [
            ElevatedButton(
              onPressed: () {
                context.goNamed('levels');
              },
              child: const Text('Levels'),
            ),
            ElevatedButton(
              onPressed: () {
                context.goNamed(
                  'game',
                  params: {'level': (widget.level + 1).toString()},
                );
              },
              child: const Text('Next'),
            ),
            ElevatedButton(
              onPressed: () {
                context.goNamed(
                  'game',
                  params: {'level': (widget.level).toString()},
                );
              },
              child: const Text('Retry'),
            ),
          ],
        ),
      ),
    );
  }
}

```

**Comment by spydon** [comment]
This should be solved in `main` and will be released as v1.9.0 soon.
